libs/designsystem/src/lib/components/dropdown/dropdown.component.ts

```diff
@@ -217,22 +230,20 @@ export class DropdownComponent
       };
       const callback: IntersectionObserverCallback = (entries) => {
         // Only apply alignment when opening:
-        if (this.state !== 'opening') {
+        if (this.state !== openState.opening) {
           return;
         }
 
+        if (this.cardAlign === cardAlignment.right) {
+          this._horizontal = 'end';
+        }
         // Cancel any pending timer to show dropdown:
         clearTimeout(this.showDropdownTimeoutId);
-
         const entry = entries[0];
         const isVisible = entry.boundingClientRect.width > 0;
         if (isVisible && entry.intersectionRatio < 1) {
-          // entry not fully showing:
-          if (entry.boundingClientRect.right > entry.rootBounds.right) {
-            // entry is cut off to the right by ${entry.boundingClientRect.right - entry.intersectionRect.right}px
-            // align to the end:
-            this._horizontal = 'end';
-          }
+          this.setCardAlignment(entry);
```

Comment:
Nice refactoring 👌 
Consider if vertical aligment should be extracted to a method as well....

Reply:
Also extracted vertical